masking_package/implementation_function.py

```diff
@@ -0,0 +1,207 @@
+					f' experiment by pressing the "{button_welcome_quit}" button. Enjoy!')
+
+# ^ in the second row, all appendices will be entered
+
+
+'''Adjust task instructions, based on the selected buttons'''
+detec_instructions.setText(f'Press "{button_detec_pres}", if a picture was shown and "{button_detec_abs}", if no picture was shown.')
+local_instructions.setText(f'Press "{button_local_left}", if the picture was shown to the left and "{button_local_right}", if the picture was shown to the right.')
+disc_instructions.setText(f'Press "{button_disc_A}", if the picture showed a(n) {disc_cat_A} and press "{button_disc_B}", if the picture showed a(n) {disc_cat_B}.')
+
+
+determine_masking_type()
+
+'''Create unique trial sequence'''
+create_trial_sequence(N_trials, N_blanks, task_local, task_disc) #see background functions
+
+
+'''Actual implementation of the experiment'''
+welcome_txt.draw()
+win.flip()
+waitKeys(keyList = [button_welcome_quit]) #display the welcome text, til people want to continue
+
+clearEvents()	#makes sure that pressing the quit button before trial onset is ignored
+
+for trial_number in range(len(N_types)): #for every trial that is implemented
+	quitkey = getKeys(keyList = [button_quit]) #check if the quit button has been pressed last trial
+	if len(quitkey) > 0: #if that's the case,
+		log.write('Experiment quit')
+		break #stop the experiment (or at least, the trials / for-loop)
+
+	if	N_types[trial_number] < 5: #if the stimuli is supposed to be shown non-centrally (indicated by trial_type index)
+		if	N_types[trial_number] % 2 == 1: #if they're supposed to be shown left (i.e., their index is an odd number)
+		    #all pictures are shown left
+			example_target.setPos(left)
+			example_foil.setPos(left)
+		else: #if they're supposed to be shown on the right
+			#all pictures are shown right
+			example_target.setPos(right)
+			example_foil.setPos(right)
+	else: #else they're showen in the center
+		#which is true for trials with indices of >= 5
+		example_target.setPos(center)
+		example_foil.setPos(center)
+
+	if fixation_cross: #implement a fixation cross, if requested
+		fix.draw()
+		win.flip()
+		wait(0.1) #fixation cross is shown for 100ms before trial onset
+		wait(0.008) #wait for half a frame to counteract refresh rate imprecisions
+
+	#forward masking / (first half of the sandwich masking)
+	if mask_forw:
+		if task_local: #if people have to determine whether the target was left or right
+			#the mask has to be drawn on both sides
+			#so that it doesn't give away the target location
+			draw_stim(mask, example_mask, right)
+			draw_stim(mask, example_mask, left)
+		else:
+			draw_stim(mask, example_mask, center)
+		win.flip()
+		wait((mask_forw_durat - 0.008)) #display the mask for the requested duration (minus half a frame)
+
+	#show stimulus
+	wait(0.008) #Note to reviewer: I'm not quite sure I remember this half-frame adjustment correctly, feel free to correct me
```

Comment:
I checked my notes and in my scripts from back then, I did it like you did, though that doesn't necessarily mean it's correct haha